mmv1/templates/terraform/decoders/monitoring_monitored_project.go.erb

```diff
@@ -12,23 +12,39 @@
 	# See the License for the specific language governing permissions and
 	# limitations under the License.
 -%>
+// terraform resource config
 config := meta.(*transport_tpg.Config)
 
-expectedName, _ := expandNestedMonitoringMonitoredProjectName(d.Get("name"), d, config)
-expectedFlattenedName := flattenNestedMonitoringMonitoredProjectName(expectedName, d, config)
-_, isNumErr := tpgresource.StringToFixed64(expectedFlattenedName.(string))
-expectProjectNumber := isNumErr == nil
+// The API returns all monitored projects
+monitoredProjects, _ := res["monitoredProjects"].([]interface{})
 
-name := res["name"].(string)
-name = tpgresource.GetResourceNameFromSelfLink(name)
+// Convert configured terraform monitored_project resource name to a ProjectNumber
+expectedProject, configProjectErr := config.NewResourceManagerClient(config.UserAgent).Projects.Get(d.Get("name").(string)).Do()
+if configProjectErr != nil {
+	return nil, configProjectErr
+}
+expectedProjectNumber := strconv.FormatInt(expectedProject.ProjectNumber, 10)
```

Comment:
Based on the error "**Parent id must be numeric.**", I wonder something needs to be changed here.

Reply:
Can you elaborate? It looks like the failure came from a "google_project" resource, I don't think a decoder for the google_monitoring_monitored_project would be part of the culprit?

Reply:
How would you approach this problem, if these tests passed previously

Reply:
I've never got these tests to pass locally. Other reviewers told me not to worry about it. It doesn't seem like these tests are very repeatable/reproducible in different environments. It does not seem to have the same issue in the CI tests.

Reply:
Do you mean that even before the change these tests still fail?

Reply:
How do we find out what the cause is?

And yes, if I revert my changes, the test still fails.

Reply:
https://github.com/GoogleCloudPlatform/magic-modules/pull/8454.
https://github.com/GoogleCloudPlatform/magic-modules/pull/8322. 

Looks like these tests passed before. 
The error is still most likely introduced by changes from this PR...

Reply:
Can we get the debug logs for what these tests currently output on the main branch?

Reply:
https://paste.googleplex.com/6652436600061952

Reply:
I'm looking into the delete_url of monitoringProject.yaml, see if we can properly change that to fit the right delete url.

Reply:
The delete works after changing the delete_url to **v1/{{name}}**